### ISSUE TITLE
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined

### ISSUE THREAD

**Issue by Hoishin** [renovatebot/renovate#10042]
**What Renovate type, platform and version are you using?**

**Renovate type**

- [ ] Hosted app
- [x] Self hosted

**Renovate platform**

- [ ] Azure DevOps (dev.azure.com)
- [ ] Azure DevOps Server (self-hosted)
- [ ] Bitbucket Cloud (bitbucket.org)
- [ ] Bitbucket Server (self-hosted)
- [ ] Gitea (gitea.com, self-hosted)
- [x] GitHub (github.com, GitHub Enterprise)
- [ ] GitLab (gitlab.com, self-hosted)

**Renovate version (if self hosting)**

I'm self hosting Renovate with version: 25.21.13 (Through docker image `renovate/renovate:25.21.13`)
Also running on Node.js 14.17.0

**Describe the bug**

- Log filled with this warn message "The \"path\" argument must be of type string. Received undefined"
- branches are not updated
- pull requests are not closed when it's supposed to

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

```
WARN: Error updating branch (repository=[deducted], branch=[deducted])
       "err": {
         "code": "ERR_INVALID_ARG_TYPE",
         "message": "The \"path\" argument must be of type string. Received undefined",
         "stack": "TypeError [ERR_INVALID_ARG_TYPE]: The \"path\" argument must be of type string. Received undefined\n    at validateString (internal/validators.js:124:11)\n    at Object.join (path.js:1110:7)\n    at Object.join (/usr/src/app/node_modules/upath/build/code/upath.js:51:33)\n    at Object.getAdditionalFiles (/usr/src/app/node_modules/renovate/lib/manager/npm/post-update/index.ts:456:35)\n    at Object.processBranch (/usr/src/app/node_modules/renovate/lib/workers/branch/index.ts:276:29)\n    at Object.writeUpdates (/usr/src/app/node_modules/renovate/lib/workers/repository/process/write.ts:38:17)\n    at Object.update (/usr/src/app/node_modules/renovate/lib/workers/repository/process/extract-update.ts:129:11)\n    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:47:17)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:112:7)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:16:22"
       }
```

</details>

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [x] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

Does not happen with previous release, 25.21.12 (docker image `renovate/renovate:25.21.12`)

**Comment by viceice** [comment]
which git version do you use?

**Comment by Hoishin** [comment]
> which git version do you use?

git version 2.31.1
this is the pre-installed git that comes with `renovate/renovate` docker image

**Comment by Hoishin** [comment]
If it's related to git, is it something to do with the docker images being updated in recent hours? I should probably have specified docker image versions rather than renovate versions:

- This bug happens on `renovate/renovate:latest`
- Doesn't happen on `renovate/renovate:25.21.12`

**Comment by p3k** [comment]
same error here since today in the morning (cest) – this is our config:

```yml
# GiLab job in `.gitlab-ci.yml`
dependencies:
  stage: update
  cache: {}
  variables:
    GITHUB_COM_TOKEN: ${GITHUB_AUTH_TOKEN}
    LOG_LEVEL: info
    RENOVATE_CONFIG_FILE: ./.gitlab/renovate.json
  before_script:
    - npm install --global yarn@=1.22.10
    - yarn global add renovate@^25
    - echo "Using $(git version) in $(which git)"
    - echo "Using Node version $(node -v) in $(which node)"
    - echo "Using Yarn version $(yarn -v) in $(which yarn)"
    - echo "Using Renovate version $(renovate -v) in $(which renovate)"
  script:
    - >
      renovate
      --token ${GITLAB_AUTH_TOKEN}
      --host-rules '[
        {
          "matchHost": "nexus.semanticlab.net",
          "password": "'${NEXUS_PASSWORD}'",
          "username": "'${NEXUS_USERNAME}'"
        }
      ]'
```

```dockerfile
# Dockerfile
FROM openjdk:11-slim

ENV LANG C.UTF-8
ENV LANGUAGE C:C.UTF-8
ENV LC_ALL C.UTF-8

RUN apt -qy update
RUN apt install -qy curl git

RUN curl -sL https://deb.nodesource.com/setup_14.x  | bash -
RUN apt -qy install nodejs

RUN apt autoremove -qy && apt autoclean -qy

CMD ["bash"]
```

```plain
# log output
Using git version 2.20.1 in /usr/bin/git
Using Node version v14.17.0 in /usr/bin/node
Using Yarn version 1.22.10 in /usr/bin/yarn
Using Renovate version 25.21.13 in /usr/local/bin/renovate
```

error message as in issue description.

**Comment by rarkins** [comment]
Related to:

```
  env.NPM_CONFIG_CACHE =
    env.NPM_CONFIG_CACHE || upath.join(config.cacheDir, './others/npm');
```

Related to: https://github.com/renovatebot/renovate/pull/9990

**Comment by rarkins** [comment]
Reverting in #10044, please downgrade to `25.21.12` in the meantime as it will take a while to test, release and build the new image.

**Comment by rarkins** [comment]
Added a fix in https://github.com/renovatebot/renovate/commit/b068ec9c0815891c5b44e75207268f9f3b1dc9fa instead, it will be in the next release (which is built automatically but may take some hours to reach Docker Hub)

**Comment by p3k** [comment]
> Added a fix in [b068ec9](https://github.com/renovatebot/renovate/commit/b068ec9c0815891c5b44e75207268f9f3b1dc9fa) instead, it will be in the next release (which is built automatically but may take some hours to reach Docker Hub)

works again with v25.21.15, thanks for the quick fix.
